src/poller.ts

```diff
@@ -0,0 +1,158 @@
+import {
+  DEFAULT_INITIAL_CONFIG_REQUEST_RETRIES,
+  DEFAULT_POLL_CONFIG_REQUEST_RETRIES,
+  POLL_JITTER_PCT,
+} from './constants';
+
+export interface IPoller {
+  start: () => Promise<void>;
+  stop: () => void;
+}
+
+// TODO: change this to a class with methods instead of something that returns a function
+
+export default function initPoller(
+  intervalMs: number,
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  callback: () => Promise<any>,
+  options?: {
+    maxPollRetries?: number;
```

Comment:
I wonder if customers would ever want to set different parameters for init polling and regular polling, but that's probably not worth figuring out right now